packages/amplify-cli/src/__tests__/utils/check-hosting.test.ts

```diff
@@ -0,0 +1,36 @@
+import { hasCdBranches } from '../../utils/check-hosting';
+
+let mockBranches = { branches: [{ some_branch_obj: 'main' }] };
+
+jest.mock('../../extensions/amplify-helpers/get-plugin-instance', () => {
+  return {
+    getPluginInstance: (_context, _plugin) => {
+      return {
+        getConfiguredAmplifyClient: (_context, _) => {
+          return Promise.resolve({
+            listBranches: _options => {
+              return {
+                promise: () => mockBranches,
+              };
+            },
+          });
+        },
+      };
+    },
+  };
+});
+
+describe('hasCdBranches', () => {
+  let mockContext = { exeInfo: { amplifyMeta: { providers: { awscloudformation: { AmplifyAppId: 'id' } } } } };
+
+  it('should return true if CD branches are present', async () => {
+    const result = await hasCdBranches(mockContext);
+    expect(result).toBe(true);
+  });
+
+  it('should return false if no CD branches are present', async () => {
+    mockBranches.branches.pop();
```

Comment:
Can you assign `mockBranches` to the value each test requires in the test itself. Otherwise, if tests are added/removed/moved around, they are a lot more likely to break.

Reply:
Good call - just pushed another commit :slightly_smiling_face: